vault/src/vaults.rs

```diff
@@ -69,26 +76,42 @@ impl<P: RelayPallet + BtcRelayPallet, B: BitcoinCoreApi + Clone> VaultTheftMonit
     async fn report_invalid(
         &self,
         vault_id: &VaultId,
-        proof: &[u8],
-        raw_tx: &[u8],
         transaction: &Transaction,
+        block_hash: BlockHash,
     ) -> Result<(), Error> {
-        tracing::info!("Found tx from vault {}", vault_id.pretty_printed());
+        tracing::debug!(
+            "Found txid {} from vault {}",
+            transaction.txid(),
+            vault_id.pretty_printed()
+        );
         // check if matching redeem or replace request
-        if self.btc_parachain.is_transaction_invalid(vault_id, raw_tx).await? {
-            tracing::info!("Transaction is invalid");
-            self.btc_parachain.report_vault_theft(vault_id, proof, raw_tx).await?;
+        let raw_tx = bitcoin::serialize(transaction); //self.bitcoin_core.get_raw_tx(&tx_id, &block_hash).await?;
+        if self.btc_parachain.is_transaction_invalid(vault_id, &raw_tx).await? {
+            tracing::info!(
+                "Detected theft by vault {} - txid {}. Reporting...",
+                vault_id.pretty_printed(),
+                transaction.txid()
+            );
+
+            let proof = self.bitcoin_core.get_proof(transaction.txid(), &block_hash).await?;
```

Comment:
Should we retry? Not expecting changes now, but maybe leave a TODO note here.

Reply:
Done, left a comment on the (renamed) `check_vault_transaction` function